lib/src/configuration.dart

```diff
@@ -133,6 +133,12 @@ class Configuration {
   /// by the [path] you  property. This property is ignored if the directory defined by [path] allow FIFO special files.
   String get fifoFilesFallbackPath => realmCore.getConfigFifoPath(this);
   set fifoFilesFallbackPath(String value) => realmCore.setConfigFifoPath(this, value);
+
+  Function(Realm realm)? get initialDataCallback => _initialDataCallback;
+  set initialDataCallback(Function(Realm realm)? value) {
+    _initialDataCallback = value;
+    realmCore.setConfigInitialDataCallback(this);
```

Comment:
We are highly inconsistent already, but I think we should try to only traffic handles to `realmCore` to separate concerns a bit more concisely.

Reply:
We did a discussion on this at the round-table. It was agreed that it is inconsistent, but not what to do about it.

There where basically two viewpoints:
1) We should use the higher level objects (RealmList, Configuration, etc) on the interface of `_RealmCore` and let the methods extract the handles they need for these.
2) We should have a clean layering, so that methods on `_RealmCore` never know about the higher level objects, hence always trafficking in handles.

The arguments was not settled, but I believe it was decided to live with this inconsistency for now, and revisist later

Reply:
Discussed offline - we're keeping it as it is. This will change in any case if/when #455 gets approved.